dao/src/main/java/se/fortnox/reactivewizard/db/ReactiveStatementFactory.java

```diff
@@ -53,29 +56,34 @@ private static void closeSilently(Connection connection) {
         }
     }
 
-    /**
-     * Create observable statement.
-     *
-     * @param args                the arguments
-     * @param connectionScheduler the scheduler
-     * @return the observable statement
-     */
-    public Object create(Object[] args, ConnectionScheduler connectionScheduler) {
-        StatementContext statementContext = new StatementContext(() -> statementFactory.create(args), connectionScheduler);
-        Flux<Object> result = Flux.create(fluxSink -> {
+    private Flux<Object> getResult(StatementContext statementContext) {
+        return Flux.create(fluxSink -> {
             try {
                 statementContext.getConnectionScheduler()
-                        .schedule(fluxSink::error, connection -> {
-                            Statement dbStatement = statementContext.getStatement();
-                            dbStatement.setFluxSink(fluxSink);
-                            executeStatement(dbStatement, connection);
-                        });
+                    .schedule(fluxSink::error, connection -> {
+                        Statement dbStatement = statementContext.getStatement();
+                        dbStatement.setFluxSink(fluxSink);
+                        executeStatement(dbStatement, connection);
+                    });
             } catch (Exception e) {
                 if (!fluxSink.isCancelled()) {
                     fluxSink.error(e);
                 }
             }
         }, FluxSink.OverflowStrategy.ERROR);
+    }
+
+    /**
+     * Create Flux statement
+     *
+     * @param args                the arguments
+     * @param connectionScheduler the scheduler
+     * @param methodReturnType    the return type of the dao method
+     * @return the Flux statement
+     */
+    public <T extends Publisher<?>> Object create(Object[] args, ConnectionScheduler connectionScheduler, Class<T> methodReturnType) {
```

Comment:
The methodReturnType is already considered when `resultConverter` is created at startup. Can the new logic be added at startup as well? Java refelection is not very performant so it is nice to avoid reflection in runtime, like below.